packages/devtools_app/lib/src/network/flutter/http_headers_tab.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:recase/recase.dart';
+
+import '../http_request_data.dart';
+
+/// This widget displays general HTTP request / response information that is
+/// contained in the headers, in addition to the standard connection information.
+class HttpRequestHeadersTab extends StatelessWidget {
+  const HttpRequestHeadersTab(this.data);
+
+  ExpansionTile _buildTile(String title, List<Widget> children) =>
+      ExpansionTile(
+        title: Text(title,
+            style: const TextStyle(fontWeight: FontWeight.bold, fontSize: 18)),
+        children: children,
+        initiallyExpanded: true,
+      );
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(builder: (context, constraints) {
+      return ListView(
+        children: <Widget>[
+          _buildTile(
+            'General',
+            <Widget>[
+              for (final entry in data.general.entries)
```

Comment:
How many entries can we expect to have?

Reply:
~20 or so from my tests, but it depends on how complex a request/response is.